homeassistant/components/rfxtrx/binary_sensor.py

```diff
@@ -261,14 +253,43 @@ def unique_id(self):
         """Return unique identifier of remote device."""
         return self._unique_id
 
-    def apply_cmd(self, cmd):
-        """Apply a command for updating the state."""
-        if cmd == self.cmd_on:
-            self.update_state(True)
-        elif cmd == self.cmd_off:
-            self.update_state(False)
-
-    def update_state(self, state):
-        """Update the state of the device."""
-        self._state = state
-        self.schedule_update_ha_state()
+    def _apply_event_lighting4(self, event):
+        """Apply event for a lighting 4 device."""
+        if self.data_bits is not None:
+            cmd = get_pt2262_cmd(event.device.device_id, self.data_bits)
+            cmd = int(cmd, 16)
+            if cmd == self.cmd_on:
+                self._state = True
+            elif cmd == self.cmd_off:
+                self._state = False
+        else:
+            self._state = True
+
+    def _apply_event_standard(self, event):
+        if event.values["Command"] in COMMAND_ON_LIST:
+            self._state = True
+        elif event.values["Command"] in COMMAND_OFF_LIST:
+            self._state = False
+
+    def _apply_event(self, event):
+        """Apply command from rfxtrx."""
+        if self.is_lighting4:
+            self._apply_event_lighting4(event)
+        else:
+            self._apply_event_standard(event)
+
+        if self.hass:
+            self.schedule_update_ha_state()
+            if self.should_fire_event:
+                fire_command_event(self.hass, self.entity_id, event.values["Command"])
```

Comment:
In the future, the entity should not fire events.

Reply:
Right. i'll move that to integration level. But breaking change that too since it will then loose the "naming" of the entity id.

Reply:
https://github.com/home-assistant/core/pull/37565